test/src/module/storage/remoteTest.c

```diff
@@ -264,6 +264,24 @@ testRun(void)
             strZ(strNewBuf(storageGetP(storageNewReadP(storagePgWrite, STRDEF("test.txt"))))), FileMissingError,
             "raised from remote-0 shim protocol: " STORAGE_ERROR_READ_MISSING, TEST_PATH "/pg256/test.txt");
 
+        // -------------------------------------------------------------------------------------------------------------------------
+        TEST_TITLE("raw read file without compression");
+
+        HRN_STORAGE_PUT(storageTest, TEST_PATH "/repo128/test.txt", contentBuf);
+
+        // Disable protocol compression in the storage object
+        ((StorageRemote *)storageDriver(storageRepo))->compressLevel = 0;
+
+        StorageRead *fileReadRaw = NULL;
+        Buffer *buffer = bufNew(bufSize(contentBuf));
+        TEST_ASSIGN(fileReadRaw, storageNewReadP(storageRepo, STRDEF("test.txt")), "new file");
+        TEST_RESULT_BOOL(ioReadOpen(storageReadIo(fileReadRaw)), true, "open read");
+        size_t size;
+        TEST_ASSIGN(size, storageReadRemote(fileReadRaw->driver, buffer, true), "read file and save returned size");
+        TEST_RESULT_UINT(size, bufUsed(buffer), "check returned size");
+        TEST_RESULT_VOID(ioReadClose(storageReadIo(fileReadRaw)), "close");
+        TEST_RESULT_BOOL(bufFull(buffer), true, "check if fill buffer");
```

Comment:
I suggest to replace this line with
```
TEST_RESULT_UINT(size, bufUsed(contentBuf), "returned size should be the same as the file size");
```
and move it closer to the previous test for "size"